tokio-trace/examples/sloggish/sloggish_subscriber.rs

```diff
@@ -70,73 +74,115 @@ pub struct SloggishSubscriber {
     stderr: io::Stderr,
     stack: Mutex<Vec<Id>>,
     spans: Mutex<HashMap<Id, Span>>,
-    events: Mutex<HashMap<Id, Event>>,
     ids: AtomicUsize,
 }
 
 struct Span {
     parent: Option<Id>,
-    kvs: Vec<(String, String)>,
+    kvs: Vec<(&'static str, String)>,
 }
 
-struct Event {
-    level: tokio_trace::Level,
-    target: String,
-    message: String,
-    kvs: Vec<(String, String)>,
+struct Event<'a> {
+    stderr: io::StderrLock<'a>,
+    comma: bool,
 }
 
-struct ColorLevel(Level);
+struct ColorLevel<'a>(&'a Level);
 
-impl fmt::Display for ColorLevel {
+impl<'a> fmt::Display for ColorLevel<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.0 {
-            Level::TRACE => Color::Purple.paint("TRACE"),
-            Level::DEBUG => Color::Blue.paint("DEBUG"),
-            Level::INFO => Color::Green.paint("INFO"),
-            Level::WARN => Color::Yellow.paint("WARN "),
-            Level::ERROR => Color::Red.paint("ERROR"),
+            &Level::TRACE => Color::Purple.paint("TRACE"),
+            &Level::DEBUG => Color::Blue.paint("DEBUG"),
+            &Level::INFO => Color::Green.paint("INFO "),
+            &Level::WARN => Color::Yellow.paint("WARN "),
+            &Level::ERROR => Color::Red.paint("ERROR"),
         }
         .fmt(f)
     }
 }
 
 impl Span {
-    fn new(parent: Option<Id>, _meta: &tokio_trace::Metadata) -> Self {
-        Self {
+    fn new(
+        parent: Option<Id>,
+        _meta: &tokio_trace::Metadata,
+        values: &tokio_trace::field::ValueSet,
+    ) -> Self {
+        let mut span = Self {
             parent,
             kvs: Vec::new(),
-        }
+        };
+        values.record(&mut span);
+        span
+    }
+}
+
+impl Record for Span {
+    fn record_i64(&mut self, field: &Field, value: i64) {
+        self.record_debug(field, &value)
```

Comment:
This seems like a common enough pattern that I wonder if we might want the `Record` trait to have default impls that call `record_debug`?

Reply:
Sounds very plausible.

Reply:
Done in 721d38f052151b08520baa92ae66a345a298097b!